roles/default/kiali-deploy/templates/kubernetes/service.yaml

```diff
@@ -12,6 +12,9 @@ metadata:
   annotations:
     kiali.io/api-spec: https://kiali.io/api
     kiali.io/api-type: rest
+{% if kiali_vars.server.web_fqdn|length != 0 and kiali_vars.server.web_schema|length != 0 %}
+    kiali.io/external-url: {{ kiali_vars.server.web_schema + '://' + kiali_vars.server.web_fqdn + ((':' + kiali_vars.server.web_port | string) if (kiali_vars.server.web_port | string | length != 0) else '') + (kiali_vars.server.web_root | default('')) }}
+{% endif %}
```

Comment:
Is this feature only for non-OpenShift environments? If not (i.e. if this works on OpenShift, too), then the same change should go here: https://github.com/kiali/kiali-operator/blob/v1.31.0/roles/default/kiali-deploy/templates/openshift/service.yaml#L15

Don't forget the server helm chart, too --> https://github.com/kiali/helm-charts/blob/v1.31.0/kiali-server/templates/service.yaml#L14

Reply:
Here is the equivalent change for Helm: https://github.com/kiali/helm-charts/pull/52

For OpenShift, I think the proper way to do it is to query the created route. It should contain the needed data, so an annotation to the Service would be redundant.

Reply:
OK. So we will only do this annotation thing for non-Openshift clusters, and the Kiali server will auto-detect using the route when on OpenShift.